server/src/internalClusterTest/java/org/opensearch/cluster/SimpleClusterStateIT.java

```diff
@@ -103,6 +103,7 @@ public void indexData() throws Exception {
         index("fuu", "buu", "1", XContentFactory.jsonBuilder().startObject().field("fuu", "fuu").endObject());
         index("baz", "baz", "1", XContentFactory.jsonBuilder().startObject().field("baz", "baz").endObject());
         refresh();
```

Comment:
I'm honestly not sure, but should be just add the "wait for segment replication" step by default in this refresh() method?

Reply:
Yes that's a good idea, but there are certain segrep/remote store tests that block replication after refresh. These kind of tests are failing [here](https://github.com/opensearch-project/OpenSearch/pull/11773#issuecomment-1906077922). To overcome this I added new `refreshWithNoWaitUntilReplica()` that doesn't wait for replica after refresh.